homeassistant/bootstrap.py

```diff
@@ -332,15 +332,17 @@ def async_enable_logging(
         not err_path_exists and os.access(err_dir, os.W_OK)
     ):
 
+        err_handler: logging.FileHandler
         if log_rotate_days:
-            err_handler: logging.FileHandler = (
-                logging.handlers.TimedRotatingFileHandler(
-                    err_log_path, when="midnight", backupCount=log_rotate_days
-                )
+            err_handler = logging.handlers.TimedRotatingFileHandler(
+                err_log_path, when="midnight", backupCount=log_rotate_days, delay=True
             )
         else:
-            err_handler = logging.FileHandler(err_log_path, mode="w", delay=True)
+            err_handler = logging.handlers.RotatingFileHandler(
+                err_log_path, backupCount=1, delay=True
+            )
 
+        err_handler.rotate(err_log_path, f"{err_log_path}.bak")
```

Comment:
Instead of indicating that it's a backup, maybe it's more clear by calling it "previous"?

```suggestion
        err_handler.rotate(err_log_path, f"{err_log_path}.previous")
```

Reply:
Why not just rename the file `home-assistant.log` to `home-assistant.log.previous` instead of using a rotating log handler?

Reply:
Changed to previous, good suggestion.

I could just rename it now, but I am working on a followup PR (as stated in the PR description), adding a "change log" service call to "developer tools" in the UI. Renaming a running log is a bit more problematic, hence the change to RotatingFileHandler. I did not want to include the service call in this PR, since it solves another problem (change log at runtime).